src/BlackSlope.Api.Common/Versioning/VersionJsonConverter.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace BlackSlope.Api.Common.Versioning
+{
+    public class VersionJsonConverter : JsonConverter<Version>
```

Comment:
One of the limitations of System.Text.Json is in Converter logic.

Normally I'd leave all this alone, but the `Version` object structure was intended to be immutable. I tried to adhere to this as much as possible.

See [this conversion guide](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to#deserialize-to-immutable-classes-and-structs)

Reply:
Can we get some unit tests added for this VersionJsonConverter.cs logic. Thanks